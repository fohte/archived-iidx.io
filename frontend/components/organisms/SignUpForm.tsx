import { Button, Form, Icon, Input } from 'antd'
import { FormComponentProps } from 'antd/lib/form'
import * as React from 'react'

import InputIcon from '@app/components/atoms/InputIcon'
import withSubmitHandling, {
  ComponentExternalProps,
  InjectedProps,
} from '@app/lib/withSubmitHandling'

export type ExternalProps = ComponentExternalProps<FormValues>

export interface Props
  extends ExternalProps,
    FormComponentProps,
    InjectedProps {}

export interface State {
  submitting: boolean
  errorMessage: string | null
}

export interface FormValues {
  username: string
  email: string
  password: string
}

const SignUpForm: React.SFC<Props> = ({
  form: { getFieldDecorator },
  errorMessage,
  handleSubmit,
  submitting,
}) => (
  <Form onSubmit={handleSubmit}>
    <Form.Item>
      {getFieldDecorator('email', {
        rules: [{ required: true, message: 'Please input your email.' }],
      })(<Input prefix={<InputIcon type="mail" />} placeholder="Email" />)}
    </Form.Item>
    <Form.Item>
      {getFieldDecorator('username', {
        rules: [{ required: true, message: 'Please input your username.' }],
      })(<Input prefix={<InputIcon type="user" />} placeholder="Username" />)}
    </Form.Item>
    <Form.Item>
      {getFieldDecorator('password', {
        rules: [{ required: true, message: 'Please input your password.' }],
      })(
        <Input
          prefix={<InputIcon type="lock" />}
          type="password"
          placeholder="Password"
        />,
      )}
    </Form.Item>
    <Form.Item>
      <Button type="primary" htmlType="submit">
        {submitting ? <Icon type="loading" /> : 'Log in'}
      </Button>
      {errorMessage}
    </Form.Item>
  </Form>
)

const WrappedNormalSignUpForm = Form.create<ExternalProps>()(
  withSubmitHandling<FormValues>()(SignUpForm),
)

export default WrappedNormalSignUpForm
