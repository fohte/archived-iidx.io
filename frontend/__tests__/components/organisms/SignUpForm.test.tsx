import * as React from 'react'
import * as renderer from 'react-test-renderer'

import SignUpForm from '@app/components/organisms/SignUpForm'

describe('SignUpForm', () => {
  it('renders correctly', () => {
    const component = renderer.create(<SignUpForm />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
