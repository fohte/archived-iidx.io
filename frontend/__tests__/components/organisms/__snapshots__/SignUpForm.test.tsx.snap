// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm renders correctly 1`] = `
<form
  className="ant-form ant-form-horizontal"
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item"
  >
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control"
      >
        <span
          className="ant-form-item-children"
        >
          <span
            className="ant-input-affix-wrapper"
          >
            <span
              className="ant-input-prefix"
            >
              <i
                className="anticon anticon-mail sc-bdVaJa eaTnkz"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="mail"
                  fill="currentColor"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 110.8V792H136V270.8l-27.6-21.5 39.3-50.5 42.8 33.3h643.1l42.8-33.3 39.3 50.5-27.7 21.5zM833.6 232L512 482 190.4 232l-42.8-33.3-39.3 50.5 27.6 21.5 341.6 265.6a55.99 55.99 0 0 0 68.7 0L888 270.8l27.6-21.5-39.3-50.5-42.7 33.2z"
                  />
                </svg>
              </i>
            </span>
            <input
              className="ant-input"
              data-__field={
                Object {
                  "name": "email",
                }
              }
              data-__meta={
                Object {
                  "name": "email",
                  "originalProps": Object {
                    "disabled": false,
                    "placeholder": "Email",
                    "prefix": <Styled(Icon)
                      type="mail"
                    />,
                    "prefixCls": "ant-input",
                    "type": "text",
                  },
                  "ref": null,
                  "rules": Array [
                    Object {
                      "message": "Please input your email.",
                      "required": true,
                    },
                  ],
                  "trigger": "onChange",
                  "validate": Array [
                    Object {
                      "rules": Array [
                        Object {
                          "message": "Please input your email.",
                          "required": true,
                        },
                      ],
                      "trigger": Array [
                        "onChange",
                      ],
                    },
                  ],
                  "valuePropName": "value",
                }
              }
              disabled={false}
              id="email"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Email"
              style={null}
              type="text"
              value=""
            />
          </span>
        </span>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
  >
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control"
      >
        <span
          className="ant-form-item-children"
        >
          <span
            className="ant-input-affix-wrapper"
          >
            <span
              className="ant-input-prefix"
            >
              <i
                className="anticon anticon-lock sc-bdVaJa eaTnkz"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="lock"
                  fill="currentColor"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 1 0-56 0z"
                  />
                </svg>
              </i>
            </span>
            <input
              className="ant-input"
              data-__field={
                Object {
                  "name": "password",
                }
              }
              data-__meta={
                Object {
                  "name": "password",
                  "originalProps": Object {
                    "disabled": false,
                    "placeholder": "Password",
                    "prefix": <Styled(Icon)
                      type="lock"
                    />,
                    "prefixCls": "ant-input",
                    "type": "password",
                  },
                  "ref": null,
                  "rules": Array [
                    Object {
                      "message": "Please input your password.",
                      "required": true,
                    },
                  ],
                  "trigger": "onChange",
                  "validate": Array [
                    Object {
                      "rules": Array [
                        Object {
                          "message": "Please input your password.",
                          "required": true,
                        },
                      ],
                      "trigger": Array [
                        "onChange",
                      ],
                    },
                  ],
                  "valuePropName": "value",
                }
              }
              disabled={false}
              id="password"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Password"
              style={null}
              type="password"
              value=""
            />
          </span>
        </span>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
  >
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control"
      >
        <span
          className="ant-form-item-children"
        >
          <button
            className="ant-btn ant-btn-primary"
            onClick={[Function]}
            type="submit"
          >
            <span>
              Sign up
            </span>
          </button>
        </span>
      </div>
    </div>
  </div>
</form>
`;
